### PR TITLE
Git 23

### LINKED ISSUE #3366: Git 23 features
### What needs to be documented?

This should be split up into individual issues, but listing them all here so we have an idea of the requirements:

### Additional context

- [ ] `pre_start` command
- [ ] Activity scripts variables namespace
- [ ] Git support for [`push-options`](https://git-scm.com/docs/git-push#Documentation/git-push.txt--oltoptiongt)
- [ ] Route information is now available in `/run/config.json` as well as `PLATFORM_ROUTES`
- [ ] Vendor: GH Application
- [ ] Vendor: Plans replaced with CPU & memory allocations. At the end of a build, user will see a recap of CPU/memory allocation for each container in the logs
- [ ] Vendor: Billing --> based on CPU/memory usage. Source operations and failed builds apply here. 
- [ ] Vendor: scaling up/down
- [ ] Vendor: scaling up/down --> disk can be weird, so local mounts = network storage
- [ ] Vendor: services and apps can be pinned to specific hosts
- [ ] Vendor: glob all files under `.vendor/*`

### PR DESCRIPTION
**Pull request by AnouckColson**
## Why

Closes #3366 

## What's changed

- [x] `pre_start` command
- [x] Activity scripts variables namespace
- [x] Git support for [`push-options`](https://git-scm.com/docs/git-push#Documentation/git-push.txt--oltoptiongt)
- [x] Route information is now available in `/run/config.json` as well as `PLATFORM_ROUTES`

All vendor features have either been addressed in Upsun docs, or not to be documented in this PR.

### CODE REVIEW

File: sites/platform/src/define-routes/_index.md [hunk 1]
```diff
@@ -383,6 +383,7 @@ all placeholders (`{default}` and `{all}`) are replaced with appropriate domain
 and any additional routes (such as redirecting HTTP to HTTPS) are created.
 This means the final generated routes differ by environment and so shouldn't be hard coded in your app.
 These routes are available in the `PLATFORM_ROUTES` environment variable as a base64-encoded JSON object.
+They are also available in your `/run/config.json` file so you can maintain a high number of routes more easily.
```
Comment (chadwcarlson) [inline sites/platform/src/define-routes/_index.md:386]:
```suggestion
They are also available within an application container from the `/run/config.json` file so you can maintain a high number of routes more easily.
```

File: sites/platform/src/environments/change-parent.md [hunk 2]
```diff
@@ -45,3 +45,12 @@ So if you have the environment `new-feature` and want to change its parent to `m
 ```
 
 {{< /codetabs >}}
+
+You can also set a parent for your environment when pushing changes to it.
+To do so, run the following command:
+
+```bash
+git push origin HEAD:{{< variable "BRANCH_NAME" >}} -o "environment.status=active"
```
Comment (chadwcarlson) [inline sites/platform/src/environments/change-parent.md:53]:
```suggestion
git push origin HEAD:{{< variable "BRANCH_NAME" >}} -o "environment.parent={{< variable "PARENT_ENVIRONMENT_NAME" >}}"
```

?

File: sites/platform/src/environments/deactivate-environment.md [hunk 3]
```diff
@@ -93,3 +93,12 @@ Run the following command:
 ```
 
 {{< /codetabs >}}
+
+You can also activate an environment when pushing changes to it.
+To do so, run the following command:
+
+```bash
+git push origin HEAD:{{< variable "BRANCH_NAME" >}} -o "environment.parent={{< variable "PARENT_ENVIRONMENT_NAME" >}}"
```
Comment (chadwcarlson) [inline sites/platform/src/environments/deactivate-environment.md:101]:
```suggestion
git push origin HEAD:{{< variable "BRANCH_NAME" >}} -o "environment.status=active"
```

Looks like these two got switched?

**Review by chadwcarlson — COMMENTED** [review]
Approved otherwise!

File: sites/platform/src/environments/_index.md [hunk 4]
```diff
@@ -271,3 +274,23 @@ title=In the Console
 {{< /codetabs >}}
 
 The environment is redeployed and becomes available for use again.
+
+## Trigger actions on `push`
+
+You can push changes to your environment and trigger the following actions at the same time:
```
Comment (remilejeune2) [inline sites/platform/src/environments/_index.md:280]: We may want to explain what are `push-option`, users may not be familiar with it and don't understand that it's standard in git.

File: sites/platform/src/environments/_index.md [hunk 5]
```diff
@@ -271,3 +274,23 @@ title=In the Console
...
+
+| Action                                     | Command                                                                                                                 |
+| ------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------- |
+| Activate the environment                   | `git push origin HEAD:<BRANCH_NAME> -o "environment.status=active"`                                    |
```
Comment (remilejeune2) [inline sites/platform/src/environments/_index.md:284]: I would remove `origin HEAD:<BRANCH_NAME>`

**Review by gilzow — DISMISSED** [review]

File: sites/platform/src/environments/_index.md [hunk 6]
```diff
@@ -271,3 +274,23 @@ title=In the Console
...
+| Set a title for the environment            | `git push origin HEAD:<BRANCH_NAME> -o "environment.title=<ENVIRONMENT_TITLE>"`       |
```
[the remark this answers is not in the document]
Comment (gilzow) [reply sites/platform/src/environments/_index.md]: we need to change `origin` to `platform` and/or make a note that the commands should be adjusted to use the remote name that they've set up for the platform-based remote location.

**Comment by gilzow** [comment]
Not working with an integration makes sense since we probably can't get github to passthru those `-o` parameters to us. But we'll need to add a warning to this section. And we can't use the `platform` remote name to get around it in an integration since the cli will update the `platform` remote to point to the integration remote address since we don't want to push directly to platform when an integration is in place.

**Review by chadwcarlson — APPROVED** [review]

**Pull request merged**